common/src/auth/auth.service.ts

```diff
@@ -81,118 +71,84 @@ export class AuthService<User extends AbstractUser> {
         })),
         tap((authResponse) => {
-          this.setToken(authResponse.token, authResponse.refreshToken, remember);
+          this.setIsAuthenticated(remember);
           this.userService.setProfile(authResponse.user);
         })
       );
   }
 
   public unauthorize(): void {
-    this.resetTokens();
+    this.resetIsAuthenticated();
     this.resetRemember();
     this.userService.resetProfile();
 
     this.router.navigate([this.authConfig.unauthenticatedRoute ?? AuthService.DEFAULT_UNAUTHENTICATED_ROUTE]);
   }
 
   public refreshToken(): Observable<HttpResponse<void>> {
+    if (this.processingTokenRefresh$) {
+      return this.processingTokenRefresh$;
+    }
+
     this.isTokenRefreshingSubject.next(true);
 
-    return this
-      .refreshToken$
+    const method = this.authConfig.refreshTokenEndpointMethod ?? 'get';
+
+    this.processingTokenRefresh$ = this.apiService[method]
+      (
+        this.authConfig.refreshTokenEndpoint ?? AuthService.DEFAULT_REFRESH_TOKEN_ENDPOINT,
+        {},
+        {
+          observe: 'response'
+        }
+      )
       .pipe(
-        take(1),
-        switchMap((refreshToken) => {
-          const data = {};
-          if (refreshToken) {
-            data[this.authConfig.refreshTokenField ?? AuthService.DEFAULT_REFRESH_TOKEN_FIELD] = refreshToken;
-          }
-
-          const method = this.authConfig.refreshTokenEndpointMethod ?? 'get';
-
-          return this.apiService[method]
-            <HttpResponse<void>>(
-              this.authConfig.refreshTokenEndpoint ?? AuthService.DEFAULT_REFRESH_TOKEN_ENDPOINT,
-              data,
-              {
-                observe: 'response'
-              }
-            )
-            .pipe(
-              tap((response) => {
-                const refreshTokenMode = this.authConfig.refreshTokenMode ?? RefreshTokenMode.HEADER;
-                const token = (refreshTokenMode === RefreshTokenMode.HEADER)
-                  ? response.headers.get('Authorization').split(' ')[1]
-                  : response.body[this.authConfig.tokenField ?? AuthService.DEFAULT_TOKEN_FIELD];
-
-                const remember = this.getRemember();
-
-                this.setToken(token, undefined, remember);
-                this.isTokenRefreshingSubject.next(false);
-              }),
-              catchError((error) => {
-                this.isTokenRefreshingSubject.next(false);
-
-                return throwError(error);
-              })
-            )
-        })
-      );
-  }
+        share(),
+        tap(() => {
+          const remember = this.getRemember();
 
-  public setToken(token: string, refreshToken?: string, remember: boolean = true): void {
-    const storage = (remember) ? localStorage : sessionStorage;
+          this.setIsAuthenticated(remember);
 
-    storage.setItem('remember', String(remember));
+          delete this.processingTokenRefresh$;

```

Comment:
@astorozhevsky 

Can we set it to `null` instead, and extend the typing of this field to `Observable<HttpResponse<void>> | null`?